src/codegen/ast-substitutions.lisp

```diff
@@ -25,150 +29,60 @@
 (deftype ast-substitution-list ()
   '(satisfies ast-substitution-list-p))
 
-(defgeneric apply-ast-substitution (subs node)
```

Comment:
I replaced this explicit generic with a call to `traverse`, emphasizing the four cases in which something actually gets done